torch/_inductor/fx_passes/post_grad.py

```diff
@@ -452,6 +455,50 @@ def is_valid_splitwithsizes_cat(match):
     return True
 
 
+
+def reinplace_scatters(graph):
+    """
+    Reinplaces scatter operations in easy cases where the node being mutated
+    is only used by the scatter (users == 1)
+
+    Also handles input mutations when there is a corresponding copy node.
```

Comment:
I don't think this will catch cases where the node being mutated only has one user, but it's an **alias** of an input. Totally fine to not support, but maybe worth calling out in the comment?

E.g.
```
@torch.compile
def f(x, indices, values):
    x1 = x.view(-1)
    x1.index_put_((indices,), values)
```

That produces this graph:
```
graph():
    %arg0_1 : [num_users=2] = placeholder[target=arg0_1]
    %arg1_1 : [num_users=1] = placeholder[target=arg1_1]
    %arg2_1 : [num_users=1] = placeholder[target=arg2_1]
    %view : [num_users=1] = call_function[target=torch.ops.aten.view.default](args = (%arg0_1, [-1]), kwargs = {})
    %index_put : [num_users=1] = call_function[target=torch.ops.aten.index_put.default](args = (%view, [%arg1_1], %arg2_1), kwargs = {})
    %view_1 : [num_users=2] = call_function[target=torch.ops.aten.view.default](args = (%index_put, [8]), kwargs = {})
    %view_2 : [num_users=0] = call_function[target=torch.ops.aten.view.default](args = (%view_1, [-1]), kwargs = {})
    %copy_ : [num_users=0] = call_function[target=torch.ops.aten.copy_.default](args = (%arg0_1, %view_1), kwargs = {})
    return ()
```

Reply:
Hmm... this is kinda tricky, will need to think more about it.

Does `copy_` always copy into an *input*? Or can it copy it into a view of an input.

Reply:
Hmm, I think `copy_()` is always guaranteed to copy into an input, yep (thanks to this line: https://github.com/pytorch/pytorch/blob/main/torch/_functorch/aot_autograd.py#L1403)